rclcpp/src/rclcpp/executor.cpp

```diff
@@ -22,12 +22,27 @@ using rclcpp::executor::Executor;
 
 Executor::Executor(rclcpp::memory_strategy::MemoryStrategy::SharedPtr ms)
 : spinning(false), interrupt_guard_condition_(rmw_create_guard_condition()),
+  waitset_(rmw_create_waitset(0)),
   memory_strategy_(ms)
 {
+  if (!waitset_) {
+    throw std::runtime_error("Failed to create waitset in Executor constructor");
+  }
+  if (!interrupt_guard_condition_) {
+    throw std::runtime_error("Failed to create interrupt gaurd condition in Executor constructor");
+  }
```

Comment:
Better, but when the `waitset` is valid but not the `interrupt_guard_condition_` it still leaks the `waitset`.